google/cloud/pubsub/samples/samples.cc

```diff
@@ -1174,6 +1173,74 @@ void CreateTopicWithSchema(google::cloud::pubsub::TopicAdminClient client,
   (std::move(client), argv.at(0), argv.at(1), argv.at(2), argv.at(3));
 }
 
+void CreateTopicWithSchemaRevisions(
+    google::cloud::pubsub::TopicAdminClient client,
+    std::vector<std::string> const& argv) {
+  //! [START pubsub_create_topic_with_schema_revisions]
+  namespace pubsub = ::google::cloud::pubsub;
+  [](pubsub::TopicAdminClient client, std::string project_id,
+     std::string topic_id, std::string schema_id, std::string const& encoding,
+     std::string const& first_revision_id,
+     std::string const& last_revision_id) {
+    auto const& schema = pubsub::Schema(project_id, std::move(schema_id));
+    auto topic = client.CreateTopic(
+        pubsub::TopicBuilder(
+            pubsub::Topic(std::move(project_id), std::move(topic_id)))
+            .set_schema(schema)
+            .set_encoding(encoding == "JSON" ? google::pubsub::v1::JSON
+                                             : google::pubsub::v1::BINARY)
+            .set_first_revision_id(first_revision_id)
+            .set_last_revision_id(last_revision_id));
+
+    // Note that kAlreadyExists is a possible error when the
+    // library retries.
+    if (topic.status().code() == google::cloud::StatusCode::kAlreadyExists) {
+      std::cout << "The topic already exists\n";
+      return;
+    } else if (topic.status().code() ==
+               google::cloud::StatusCode::kInvalidArgument) {
+      std::cout
+          << "The first revision id must be older than the last revision id.\n";
+      return;
+    }
+    if (!topic) throw std::move(topic).status();
+
+    std::cout << "The topic was successfully created: " << topic->DebugString()
+              << "\n";
+  }
+  //! [END pubsub_create_topic_with_schema_revisions]
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2), argv.at(3),
+   argv.at(4), argv.at(5));
+}
+
+void UpdateTopicSchema(google::cloud::pubsub::TopicAdminClient client,
+                       std::vector<std::string> const& argv) {
+  //! [START pubsub_update_topic_schema]
+  namespace pubsub = ::google::cloud::pubsub;
+  [](pubsub::TopicAdminClient client, std::string project_id,
+     std::string topic_id, std::string const& first_revision_id,
+     std::string const& last_revision_id) {
+    auto topic = client.UpdateTopic(
+        pubsub::TopicBuilder(
+            pubsub::Topic(std::move(project_id), std::move(topic_id)))
+            .set_first_revision_id(first_revision_id)
+            .set_last_revision_id(last_revision_id));
+
+    if (topic.status().code() == google::cloud::StatusCode::kInvalidArgument) {
```

Comment:
There are a multitude of reasons why we may get this error, just skip this `if()` I think.

Reply:
Removed the if